### PR TITLE
DOC: update pandas.DatetimeIndex.to_period docstring(Nairobi)

### PR DESCRIPTION
**Pull request by Denniskamau**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [ x] PR title is "DOC: update the pandas.DatetimeIndex.to_period docstring"
- [ x] The validation script passes: `scripts/validate_docstrings.py pandas.DatetimeIndex.to_period `
- [ x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ x] The html version looks good: `python doc/make.py --single pandas.DatetimeIndex.to_period `
- [ x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```

################################################################################
################## Docstring (pandas.DatetimeIndex.to_period) ##################
################################################################################

Cast to PeriodIndex at a particular frequency.

Converts DatetimeIndex to PeriodIndex.

Parameters
----------
freq : string or pandas offset object, optional
    One of pandas date offset string or corresponding objects.

Returns
-------
period : DatetimeIndex

Examples
--------
>>> df = pd.DataFrame({"y": [1,2,3]},
...                   index=pd.to_datetime(["2000-03-31 00:00:00",
...                                         "2000-05-31 00:00:00",
...                                         "2000-08-31 00:00:00"]))
>>> df.index = df.index.to_period("M")
>>> df
            y
2000-03 1
2000-05 2
2000-08 3

See also
--------
pandas.PeriodIndex: Immutable ndarray holding ordinal values
pandas.DatetimeIndex.to_pydatetime: Return DatetimeIndex as object

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DatetimeIndex.to_period" correct. :)

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
 
     def to_period(self, freq=None):
         """
-        Cast to PeriodIndex at a particular frequency
+        Cast to PeriodIndex at a particular frequency.
+
+        Converts timestamps index to period.
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: What do you mean by this? Oh, I think I see. How about `Converts a DatetimeIndex to a PeriodIndex`.

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+
+        Parameters
+        ----------
+        freq : string or pandas offset object,optional
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: space before `optional`.

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+            One of pandas date offset string or corresponding objects.
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Could you include a link to the table of alias. Something like

    :ref:`offset aliases <timeseries.offset_aliases>`

File: pandas/core/indexes/datetimes.py [hunk 4]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+
+        Returns
+        -------
+        period: period[D]
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: period : DatetimeIndex

(need the space between the two).

File: pandas/core/indexes/datetimes.py [hunk 5]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({"y": [1,2,3]},
+        ...                     index=pd.to_datetime(["2000-03-31 00:00:00",
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Make 'index' right below the `{`.

File: pandas/core/indexes/datetimes.py [hunk 6]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+        ...                                           "2000-05-31 00:00:00",
+        ...                                           "2000-08-31 00:00:00"]))
+        >>> df.index = df.index.to_period("M")
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: And maybe just assign `index = pd.to_datetime([...])`, no `df`.


Then `index.to_period("M")` and show the output.

File: pandas/core/indexes/datetimes.py [hunk 7]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
...
+
+        See also
+        --------
+        Index: base pandas index type
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Remove Index. Add pandas.PeriodIndex pandas.DatetimeIndex.to_pydatetime

File: pandas/core/indexes/datetimes.py (hunk 1 above)
Comment (Denniskamau) [reply pandas/core/indexes/datetimes.py]: Sure.I have made the changes.Thanks

File: pandas/core/indexes/datetimes.py [hunk 8]
```diff
@@ -1057,7 +1057,30 @@ def to_pydatetime(self):
 
     def to_period(self, freq=None):
         """
-        Cast to PeriodIndex at a particular frequency
+        Cast to PeriodIndex at a particular frequency.
+
+        Converts DatetimeIndex  to periodIndex.
```
Comment (jreback) [inline pandas/core/indexes/datetimes.py]: extra space, can you capitalize PeriodIndex

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by TomAugspurger** [comment]
Thanks @Denniskamau! I added an example with the frequency being inferred.

**Pull request merged**